client/finality-grandpa/src/environment.rs

```diff
@@ -1173,7 +1173,7 @@ where
 	debug!(target: "afg", "Finding best chain containing block {:?} with number limit {:?}", block, limit);
 
 	let result = match select_chain.finality_target(block, None).await {
-		Ok(Some(best_hash)) => {
+		Ok(best_hash) => {
 			let best_header = client
 				.header(BlockId::Hash(best_hash))?
 				.expect("Header known to exist after `finality_target` call; qed");
```

Comment:
I think this may no longer be true, there is a possibility that `SelectChain::finality_target` just returned the passed value without checking the DB at all (in practice it shouldn't happen as we can only have previously finalized something that already existed in the database). Either way we should not call `expect` here and instead handle the case where we get `None` from this call. Something like this should work: 

```rust
let best_header = match client.header(BlockId::Hash(best_hash))? {
    Some(header) => header,
    None => {
        debug!(target: "afg", "Couldn't find target block from `finality_target`: {:?}", best_hash);
        return Ok(None);
    },
};
```